src/ui/components/shared/LoadingScreen.tsx

```diff
@@ -82,46 +80,47 @@ const phrases = [
   "Applying middleware to the Fourth Dimension...",
   "Preventing cross-origin requests to the past...",
   "Running Cron job for temporal recalibration...",
-  "Loading Schrödinger's CSS (it's both formatted and not until observed)...",
+  "Loading Schrödinger's CSS (both formatted and not until observed)...",
   "Wrapping dimensions in higher-order components...",
   "Throwing promise to future self (hope they catch it)...",
   "Flattening the time-space array...",
-  "Precompiling 4th dimensional SASS...",
   "Chrono-optimizing the runtime environment...",
   "Checking quantum entanglement of state variables...",
   "Compiling tachyonic antitelephone scripts...",
-  "Ensuring the time-travel GC isn't causing a stack overflow...",
 ];
 
 function LoadingScreen({
   uploading,
   awaitingSourcemaps,
   fallbackMessage,
-  stalledTimeout = 15000,
-}: PropsFromRedux & { fallbackMessage: string; stalledTimeout?: number }) {
+  isProcessed,
+  stalledTimeout = 20000,
+}: PropsFromRedux & { fallbackMessage: string; isProcessed?: boolean; stalledTimeout?: number }) {
```

Comment:
`isProcessed` should be a _required_ prop to prevent the loading screen from being rendered with an invalid message.
```suggestion
}: PropsFromRedux & { fallbackMessage: string; isProcessed: boolean; stalledTimeout?: number }) {
```

Reply:
We might be rendering this loader in places where we don't yet have the recording (so we can't know that) in which case I think maybe it would be appropriate to make a higher level change:

1. Move the message logic out of the loading component and make it a prop that's passed in<sup>1</sup> and remove the `isProcessed` prop. (We could make this a custom hook (e.g. `useLoadingMessage(isProcessed: boolean): string`) to encapsulate it a bit more, especially if we need to use it in more than one place.)

```ts
function LoadingScreen({ message }: { message: string }) {
```

2. Pass an empty string (`""`) in for places like `pages/_app.tsx` (where we don't have any info, and we only expect the loader to show for a brief moment)

Reply:
Hey @jonbell-lot23 The change in 0249270 is not the full thing I mentioned above.